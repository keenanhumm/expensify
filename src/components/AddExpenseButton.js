import React from 'react';
import { NavLink } from 'react-router-dom';

export default () => (
    <div className="container">
        <NavLink to="/create" className="nav-link" activeStyle={{
            color: 'white',
            fontWeight: 'bold',
            textDecoration: 'none'
        }}>
            <img src="images/add.gif" height="40px" width="40px" /> New Expense
        </NavLink>
    </div>
);