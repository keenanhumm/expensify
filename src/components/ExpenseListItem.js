import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';


export const ExpenseListItem = ({id, description, amount, createdAt}) => (
    <div>
        <Link to={`/edit/${id}`}> 
            <h3>{description}</h3>
        </Link>
        <p>Amount: ${amount/ 100}</p>
        <p>Created on {moment(createdAt).format('MMM DD YYYY')} at {moment(createdAt).format('hh:mma')}</p>
    </div>
);  

export default ExpenseListItem;