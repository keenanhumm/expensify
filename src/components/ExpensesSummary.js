import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expensesCount, expensesTotal }) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
    const formattedAmount = numeral(expensesTotal / 100).format('$0, 0.00');

    return (
        <div>
            <h3>
                Viewing {expensesCount} {expenseWord}, with a grand total of {formattedAmount}.
            </h3>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: expensesTotal(visibleExpenses)
    }
};

export default connect(mapStateToProps)(ExpensesSummary);