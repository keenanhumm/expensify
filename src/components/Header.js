import React from 'react';
import { NavLink } from 'react-router-dom';
import { Image, Button } from 'react-bootstrap';

const Header = () => {
    return (
        <header>
            <div className="container">
                <NavLink to="/" className="nav-link" exact={true}><Image src="images/biglogo.png" responsive /></NavLink>
            </div>
        </header>
    );

};

export default Header;