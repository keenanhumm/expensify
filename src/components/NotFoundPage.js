import React from 'react';
import { Link } from 'react-router-dom';
export const NotFoundPage = () => (
    <div>
        <h1>404! Page Not Found</h1>
        <h2><Link to="/">Go Home</Link></h2>
    </div>
);

export default NotFoundPage;