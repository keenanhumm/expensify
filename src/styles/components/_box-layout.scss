.box-layout {
    background: url('/images/city-light.jpg');
    height: 10vh;
    height: 100vh;
}