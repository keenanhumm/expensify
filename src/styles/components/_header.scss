.big-logo {
    width: 100%;
}
.nav-bar {
    color: blue;
}