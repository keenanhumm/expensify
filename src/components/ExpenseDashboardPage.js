import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';
import AddExpenseButton from './AddExpenseButton';

export const ExpenseDashboardPage = () => (
    <div>
        <AddExpenseButton />
        <ExpenseListFilters />
        <div className="box-layout">
            <ExpensesSummary />
            <ExpenseList /> 
        </div>
       
    </div>
);


export default ExpenseDashboardPage;