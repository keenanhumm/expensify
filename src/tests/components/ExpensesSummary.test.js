import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correctly render expenses summary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expenseTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render expenses summary with more than one expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={23} expenseTotal={23212321} />);
    expect(wrapper).toMatchSnapshot();
});
