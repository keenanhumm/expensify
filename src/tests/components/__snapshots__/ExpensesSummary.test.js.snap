// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render expenses summary with 1 expense 1`] = `
<div
  className="container"
>
  <h3>
    Viewing 
    1
     
    expense
    , with a grand total of 
    $0.00
    .
  </h3>
</div>
`;

exports[`should correctly render expenses summary with more than one expense 1`] = `
<div
  className="container"
>
  <h3>
    Viewing 
    23
     
    expenses
    , with a grand total of 
    $0.00
    .
  </h3>
</div>
`;
